### PR TITLE
feat(bulk-import): allow user to select repositories from side panel

### PR DESCRIPTION
**Pull request by ciiay**
For [RHIDP-1665](https://issues.redhat.com/browse/RHIDP-1665): Allow users to select repositories from organization

Updated screen recording(updated on **4/16**):

https://github.com/janus-idp/backstage-plugins/assets/26255262/c0461e5b-459c-4960-a6d3-36dce17d5db6

### CODE REVIEW

**Comment by invincibleJai** [comment]
https://github.com/janus-idp/backstage-plugins/pull/1357 has been merged now, can do rebase.

**Comment by debsmita1** [comment]
A few observations:
- Selected repositories column cannot be sorted
- When selecting repositories from an org, if I uncheck all the selected repos, the "Select" button gets disabled.
- The check icon before "Ready" is missing

**Comment by ciiay** [comment]
Hi @ShiranHi , @debsmita1 and I have a question about the Ready status of catalog-info.yaml. I have the below two display types and one proposal. If the concept of the proposal makes sense to you, we can also work on a better style of how to display it. Let us know how you think, thank you in advance.

_Type 1_:
Key points: Only show the number of selected repos when part of the repos selected. Always display Check icon for Ready status.
rhidp_1665_option1

_Type 2_:
Key points: Using Check icon as a indicator of All repos of this row are selected. Hide Check icon when part of the repos are selected.
rhidp_1665_option2

_Proposal_:
Key points: Ready status always comes with a Check icon. Using a "/" to show what number out of the total repos are selected. 
rhidp_1665_option3

**Comment by ShiranHi** [comment]
> Hi @ShiranHi , @debsmita1 and I have a question about the Ready status of catalog-info.yaml. I have the below two display types and one proposal. If the concept of the proposal makes sense to you, we can also work on a better style of how to display it. Let us know how you think, thank you in advance.
> 
> _Type 1_: Key points: Only show the number of selected repos when part of the repos selected. Always display Check icon for Ready status. rhidp_1665_option1
> 
> _Type 2_: Key points: Using Check icon as a indicator of All repos of this row are selected. Hide Check icon when part of the repos are selected. rhidp_1665_option2
> 
> _Proposal_: Key points: Ready status always comes with a Check icon. Using a "/" to show what number out of the total repos are selected. rhidp_1665_option3

I love this idea, let's write the number of repository selected x/y (for example 3/10). Thank you for bringing this up! 

About the side panel, can we use the basic table style? the rows are too high

**Review by debsmita1 — CHANGES_REQUESTED** [review]

File: plugins/bulk-import/src/components/AddRepositories/AddRepositoriesDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            title={title}
+            setSearchString={setSearchString}
+            selectedReposFromDrawer={selectedReposID}
+            selectedRepositoriesFormData={selectedRepositoriesFormData}
+            activeOrganization={data}
+          />
+          <RepositoriesTable
+            searchString={searchString}
+            selectedOrgRepos={selectedReposID}
+            updateSelectedReposInDrawer={handleCallback}
```
[the remark this answers is not in the document]
Comment (ciiay) [reply plugins/bulk-import/src/components/AddRepositories/AddRepositoriesDrawer.tsx:107]: Good catch. I wanted to use `updateSelectedReposInDrawer` but forgot to update the declaration.

**Comment by ciiay** [comment]
Hi @debsmita1 Thanks for the review. I updated the PR and the drawer style for a small screen display. Here's the screenshot and it's scrollable too.
rhidp_1665_mobile_view

File: plugins/bulk-import/src/components/AddRepositories/AddRepositoriesDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      open={open}
+      variant="temporary"
+      className={classes.drawerPaper}
+    >
+      <Container
+        style={{
+          padding: '20px',
+          height: '100%',
+          display: 'flex',
+          flexDirection: 'column',
```
Comment (debsmita1) [inline plugins/bulk-import/src/components/AddRepositories/AddRepositoriesDrawer.tsx:89]: move this to useStyles

File: plugins/bulk-import/src/utils/repository-utils.tsx [hunk 3]
```diff
@@ -113,24 +148,39 @@ export const getNewSelectedRepositories = (
...
-    return data.repositories?.length || 0;
-  }
-  return (
-    data.organizations?.reduce((acc, org) => {
-      const repos = acc + (org.selectedRepositories || 0);
-      return repos;
-    }, 0) || 0
-  );
+  return data.repositories?.length || 0;
+  // if (data.repositoryType === 'repository') {
```
Comment (debsmita1) [inline plugins/bulk-import/src/utils/repository-utils.tsx:165]: Remove the commented lines

**Review by debsmita1 — CHANGES_REQUESTED** [review]

**Comment by ShiranHi** [comment]
> Hi @ShiranHi I have updated this PR to address the table row height issue. You can confirm the style on the updated screen recording. Thank you.

Looks much better, thank you for this. 
I just saw that the padding between the checkbox and the name is too big, can we reduce it so it will look like in the design ([Figma](https://www.figma.com/file/09Iltd57RiuqOtKWkjy8MH/Bulk-import?type=design&node-id=74%3A10076&mode=design&t=sODyyPnFY9s3WaFK-1))?

image-1

File: plugins/bulk-import/src/utils/repository-utils.tsx [hunk 4]
```diff
@@ -14,10 +14,13 @@ import {
...
+  isDrawer: boolean = false,
+  selectedRepositories?: number,
 ) => {
   if (isItemSelected) {
     return (
       <span>
+        {selectedRepositories || ''}
         <ReadyIcon
           color="success"
           style={{ verticalAlign: 'bottom', paddingTop: '7px' }}
```
Comment (debsmita1) [inline plugins/bulk-import/src/utils/repository-utils.tsx:26]: Can you update this to

```suggestion
          style={{ verticalAlign: 'sub', paddingTop: '7px' }}
```

**Review by debsmita1 — CHANGES_REQUESTED** [review]
We do not want to show catalog-info status in the side panel, we want to show it only on the selected repositories table

Screenshot 2024-04-18 at 4 04 33 PM


Figma:

Screenshot 2024-04-18 at 4 05 08 PM

**Pull request merged**